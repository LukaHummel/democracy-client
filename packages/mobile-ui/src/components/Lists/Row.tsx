import React from 'react';
import styled from 'styled-components/native';

const RowWrapper = styled.TouchableHighlight.attrs({
  underlayColor: 'rgba(68, 148, 211, 0.1)',
})`
  padding-vertical: 14;
  padding-left: 8;
  padding-right: 14;
  border-bottom-color: #c8c7cc;
  border-bottom-width: 1;
`;

interface Props {
  onPress: () => void;
}

const Row: React.FC<Props> = ({ children, onPress }) => (
  <RowWrapper onPress={onPress}>{children}</RowWrapper>
);

export { Row };
