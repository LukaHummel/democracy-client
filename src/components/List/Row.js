import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

const RowWrapper = styled.View`
  border-bottom-width: 1;
`;

const Title = styled.Text``;
const Subtitle = styled.Text`
  color: grey;
`;
const Date = styled.Text`
  color: green;
`;
const Votes = styled.Text`
  color: red;
`;

const Row = ({
  title, subtitle, date, votes,
}) => (
  <RowWrapper>
    <Title>{title}</Title>
    <Subtitle>{subtitle}</Subtitle>
    <Date>{date}</Date>
    <Votes>{votes}</Votes>
  </RowWrapper>
);

Row.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  votes: PropTypes.number.isRequired,
};

export default Row;
