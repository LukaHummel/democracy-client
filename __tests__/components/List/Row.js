import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Row from '../../../src/components/List/Row';

describe('test components/List/Row', () => {
  it('render row', () => {
    this.onpress = () => {};
    const votes = 123;
    const tree = renderer
      .create(<Row title="test" subtitle="test2" date="test3" votes={votes} onPress={this.onpress} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
