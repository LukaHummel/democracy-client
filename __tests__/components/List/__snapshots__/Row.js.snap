// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test components/List/Row render row 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#c8c7cc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 27,
          "paddingRight": 10,
          "paddingTop": 10,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#030303",
              "fontSize": 17,
              "letterSpacing": -0.41,
            },
            undefined,
          ]
        }
      >
        test
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#8f8e94",
              "fontSize": 15,
              "letterSpacing": -0.24,
              "paddingTop": 5,
            },
            undefined,
          ]
        }
      >
        test2
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "space-between",
            "width": 48,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "width": 48,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "grey",
                "fontSize": 12,
              },
              Array [
                Object {
                  "fontSize": 45,
                  "marginBottom": -15,
                  "marginTop": -10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          Ôèò
        </Text>
        <Text
          accessible={true}
          adjustsFontSizeToFit={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#8f8e94",
                "fontSize": 18,
                "letterSpacing": -0.29,
              },
              undefined,
            ]
          }
        >
          123
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "green",
            },
            undefined,
          ]
        }
      >
        3:13
      </Text>
    </View>
  </View>
</View>
`;
